cns/restserver/internalapi.go

```diff
@@ -151,6 +152,37 @@ func (service *HTTPRestService) SyncNodeStatus(dncEP, infraVnet, nodeID string,
 	return
 }
 
+// SyncHostNCVersion will check NC version from NMAgent and save it as host version in container status.
+// If NMAgent updated, CNS will refresh the pending programming IP status.
+func (service *HTTPRestService) SyncHostNCVersion(channelMode string) {
+	for i, containerstatus := range service.state.ContainerStatus {
+		// Will open a separate PR to convert all the NC version related variable to int. Change from string to int is a pain.
+		hostVersion, err := strconv.Atoi(containerstatus.HostVersion)
+		if err != nil {
+			log.Errorf("Received err when chagne containerstatus.HostVersion %s to int, err msg %v", containerstatus.HostVersion, err)
+			return
+		}
+		vmVersion, err := strconv.Atoi(containerstatus.VMVersion)
+		if err != nil {
+			log.Errorf("Received err when chagne containerstatus.VMVersion %s to int, err msg %v", containerstatus.VMVersion, err)
+			return
+		}
+		// host NC version is the NC version from NMAgent, if it's already keep up with NC version exist in VM, no update needed.
+		if hostVersion >= vmVersion {
+			continue
+		} else {
+			newHostNCVersion := service.imdsClient.GetNetworkContainerInfoFromHostWithoutToken()
```

Comment:
I see GetNetworkContainerInfoFromHostWithoutToken has a temp implementation at the moment which returns max int value. Are you planning on changing the implementation as a part of this PR? If not, this code will set the newHostVersion to max int value.
Also, I believe you would need to check if the new API ( without the token ) is supported by the nmagent on this node, if yes, call the api without the token if not, call the api which needs the token. isNCWaitingForUpdate func has the implementation to call with the token.

Reply:
I'm not planning to put implementation in this PR. Yes, it will return max int. Return max int will avoid breaking current logic.